src/value/ser.rs

```diff
@@ -387,25 +598,40 @@ impl serde::ser::SerializeStruct for SerializeMap {
     where
         T: Serialize,
     {
-        try!(serde::ser::SerializeMap::serialize_key(self, key));
-        serde::ser::SerializeMap::serialize_value(self, value)
+        match *self {
+            SerializeMap::Map { .. } => {
+                try!(serde::ser::SerializeMap::serialize_key(self, key));
+                serde::ser::SerializeMap::serialize_value(self, value)
+            },
+            #[cfg(feature = "arbitrary_precision")]
+            SerializeMap::Number { ref mut out_value } => {
+                if key == SERDE_STRUCT_FIELD_NAME {
+                    value.serialize(NumberValueEmitter(&mut *out_value))
+                } else {
+                    Err(invalid_number())
+                }
+            },
+        }
     }
 
-    fn end(self) -> Result<Value, Error> {
-        serde::ser::SerializeMap::end(self)
+    fn end(self) -> Result<Self::Ok, Self::Error> {
+        match self {
+            SerializeMap::Map { .. } => serde::ser::SerializeMap::end(self),
+            #[cfg(feature = "arbitrary_precision")]
+            SerializeMap::Number { out_value, .. } => Ok(out_value.expect("number value was not emitted")),
+        }
     }
 }
 
 impl serde::ser::SerializeStructVariant for SerializeStructVariant {
     type Ok = Value;
     type Error = Error;
 
-    fn serialize_field<T: ?Sized>(&mut self, key: &'static str, value: &T) -> Result<(), Error>
+    fn serialize_field<T: ?Sized>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
```

Comment:
I would prefer not to make this change in this PR. The existing code is inconsistent but let's leave it to a later PR to pick one usage and standardize everywhere.

Reply:
Okay, sure. I'll leave that to someone else. ;-)

Reply:
Please revert this and similar unnecessary formatting changes.

Reply:
My bad, I thought this was in code I introduced for some reason. Reverted, and left the inconsistency as-is.